mobius-android/src/test/java/com/spotify/mobius/android/MobiusLoopViewModelTest.java

```diff
@@ -0,0 +1,156 @@
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestViewEffectHandler;
+import com.spotify.mobius.functions.Consumer;
+import com.spotify.mobius.runners.ImmediateWorkRunner;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MobiusLoopViewModelTest {
+
+  private final List<TestEvent> recordedEvents = new LinkedList<>();
+  private final List<TestModel> recordedModels = new LinkedList<>();
+  private final Update<TestModel, TestEvent, TestEffect> updateFunction =
+      (model, event) -> {
+        recordedModels.add(model);
+        recordedEvents.add(event);
+        return Next.noChange();
+      };
+  @Rule public InstantTaskExecutorRule rule = new InstantTaskExecutorRule();
+  private MobiusLoopViewModel<TestModel, TestEvent, TestEffect, TestViewEffect> underTest;
+  private TestViewEffectHandler<TestEvent, TestEffect, TestViewEffect> testViewEffectHandler;
+
+  private FakeLifecycleOwner fakeLifecycle;
+  private RecordingObserver<TestModel> recordingModelObserver = new RecordingObserver<>();
+  private RecordingObserver<TestViewEffect> recordingForegroundViewEffectObserver =
+      new RecordingObserver<>();
+  private RecordingObserver<Iterable<TestViewEffect>> recordingBackgroundEffectObserver =
+      new RecordingObserver<>();
+
+  @Before
+  public void setUp() {
+    fakeLifecycle = new FakeLifecycleOwner();
+    recordedEvents.clear();
+    recordedModels.clear();
+    testViewEffectHandler = null;
+    recordingModelObserver.clearValues();
+    recordingForegroundViewEffectObserver.clearValues();
+    recordingBackgroundEffectObserver.clearValues();
+    //noinspection Convert2MethodRef
+    underTest =
+        new MobiusLoopViewModel<>(
+            (Consumer<TestViewEffect> consumer) -> {
+              testViewEffectHandler = new TestViewEffectHandler<>(consumer);
+              return Mobius.loop(updateFunction, testViewEffectHandler)
+                  .eventRunner(ImmediateWorkRunner::new)
+                  .effectRunner(ImmediateWorkRunner::new);
+            },
+            new TestModel("initial model"),
+            (TestModel model) -> First.first(model),
+            new ImmediateWorkRunner(),
+            100);
+    underTest.getModels().observe(fakeLifecycle, recordingModelObserver);
+    underTest
+        .getViewEffects()
+        .setObserver(
+            fakeLifecycle,
+            recordingForegroundViewEffectObserver,
+            recordingBackgroundEffectObserver);
+  }
+
+  @Test
+  public void testViewModelgetModelAtStartIsInitialModel() {
+    fakeLifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    assertThat(underTest.getModel().name, equalTo("initial model"));
+  }
+
+  @Test
+  public void testViewModelSendsEventAndModelIntoLoop() {
+    fakeLifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+    testViewEffectHandler.sendEvent(new TestEvent("an initial event"));
```

Comment:
This looks a little odd to me - what is this test proving? The `sendEvent` method doesn't really make sense in a real view effect handler, AFAIU? This seems to just verify that when you send an event to a mobius loop, it gets handled? 

I think it would be easier for me to understand the purpose of this test if it had a more explanatory name; I like given-when-then as  a way of clarifying that.  Also a name that describes the desired behaviour instead of what the test does - it's clear from the code that the test sends an event, but why does it do that and what's expected?

Sorry if this comment is unclear, I'm not sure how to phrase it since I don't quite understand what the intent is.

Reply:
Hm, the point of this was to make sure the initial model passed into the ViewModel is correctly given to the loop, though the only way I could verify that the ViewModel passes it into the loop was to emit an event so the Logic function could get the model. 

I just thought of possibly another way of testing the model I can try to change the test maybe (by testing that the subscribed model recording observer has the right value at start)

Reply:
I removed the test and cleaned up the file a bit. Instead added an assert for the Model observer in the initialization test above this one.